packages/grafana-e2e/src/support/types.ts

```diff
@@ -5,10 +5,12 @@ import { fromBaseUrl } from './url';
 
 export type VisitFunction = (args?: string) => Cypress.Chainable<Window>;
 export type E2EVisit = { visit: VisitFunction };
-export type E2EFunction = (text?: string) => Cypress.Chainable<JQuery<HTMLElement>>;
+export type E2EFunction = ((text?: string, options?: CypressOptions) => Cypress.Chainable<JQuery<HTMLElement>>) &
+  E2EFunctionWithOnlyOptions;
+export type E2EFunctionWithOnlyOptions = (options?: CypressOptions) => Cypress.Chainable<JQuery<HTMLElement>>;
 
 export type TypeSelectors<S> = S extends StringSelector
-  ? E2EFunction
+  ? E2EFunctionWithOnlyOptions
```

Comment:
I _think_ when we have a selector which is only a string .e.g.
```
VisualizationTab: {
          legendSection: 'Legend section',
},
```
we would only ever want to pass in the `options` object and not pass in `text` at all